Linux build fails with CMakeLists.txt error – target "disk_space_2" not built

## Description
When running a Flutter project on Linux with `disk_space_2` (v1.0.10), the build fails with a CMake error. The plugin works fine on both Android and iOS. It does not affect the project startup on macOS, but fails to start the project directly under Ubuntu.

## Steps to Reproduce
1. Create a new Flutter project.
2. Add `disk_space_2: ^1.0.10` to `pubspec.yaml`.
3. Run `flutter run -d linux`.

## Error Output
CMake Error at flutter/ephemeral/.plugin_symlinks/disk_space_2/linux/CMakeLists.txt:7 (target_compile_options):
Cannot specify compile options for target "disk_space_2" which is not built
by this project.
Error: Unable to generate build files

## Environment
- disk_space_2 version: 1.0.10
- Flutter version: flutter_3.35.2
- OS:  Ubuntu Studio 24.04 / KDE Plasma 5.27.12
- Dart SDK version: 3.9.2 (stable) (Wed Aug 27 03:49:40 2025 -0700) on "linux_x64

## generated_plugin_registrant.cc
```cc
//
//  Generated file. Do not edit.
//

// clang-format off

#include "generated_plugin_registrant.h"

#include <disk_space_2/disk_space_2_plugin.h>
#include <tray_manager/tray_manager_plugin.h>

void fl_register_plugins(FlPluginRegistry* registry) {
  g_autoptr(FlPluginRegistrar) disk_space_2_registrar =
      fl_plugin_registry_get_registrar_for_plugin(registry, "DiskSpace_2Plugin");
  disk_space_2_plugin_register_with_registrar(disk_space_2_registrar);
  g_autoptr(FlPluginRegistrar) tray_manager_registrar =
      fl_plugin_registry_get_registrar_for_plugin(registry, "TrayManagerPlugin");
  tray_manager_plugin_register_with_registrar(tray_manager_registrar);
}


```

--- a/linux/CMakeLists.txt
+++ b/linux/CMakeLists.txt
@@ -36,7 +36,7 @@ target_include_directories(${PLUGIN_NAME} PRIVATE
   ${GTK_INCLUDE_DIRS}
 )
 
-target_compile_options(${PLUGIN_NAME} PRIVATE ${GTK_CFLAGS_OTHER})
+target_compile_options(${PLUGIN_NAME} PRIVATE ${GTK_CFLAGS} ${GTK_CFLAGS_OTHER})
 
 target_link_libraries(${PLUGIN_NAME} PRIVATE
   flutter

--- a/linux/CMakeLists.txt
+++ b/linux/CMakeLists.txt
@@ -1,49 +1,49 @@
-# The Flutter tooling requires that developers have CMake 3.10 or later
-# installed. You should not increase this version, as doing so will cause
-# the plugin to fail to compile for some customers of the plugin.
 cmake_minimum_required(VERSION 3.10)
 
-# Project-level configuration.
 set(PROJECT_NAME "disk_space_2")
 project(${PROJECT_NAME} LANGUAGES CXX)
 
-# This value is used when generating builds using this plugin, so it must
-# not be changed.
 set(PLUGIN_NAME "disk_space_2_plugin")
 
-# Any new source files that you add to the plugin should be added here.
+# Source files
 list(APPEND PLUGIN_SOURCES
   "disk_space_2_plugin.cc"
 )
 
-# Define the plugin library target. Its name must not be changed (see comment
-# on PLUGIN_NAME above).
+# Create the shared library
 add_library(${PLUGIN_NAME} SHARED
   ${PLUGIN_SOURCES}
 )
 
-# Apply a standard set of build settings that are configured in the
-# application-level CMakeLists.txt. This can be removed for plugins that want
-# full control over build settings.
+# Flutter standard settings
 apply_standard_settings(${PLUGIN_NAME})
 
-# Symbols are hidden by default to reduce the chance of accidental conflicts
-# between plugins. This should not be removed; any symbols that should be
-# exported should be explicitly exported with the FLUTTER_PLUGIN_EXPORT macro.
 set_target_properties(${PLUGIN_NAME} PROPERTIES
   CXX_VISIBILITY_PRESET hidden)
+
 target_compile_definitions(${PLUGIN_NAME} PRIVATE FLUTTER_PLUGIN_IMPL)
 
-# Source include directories and library dependencies. Add any plugin-specific
-# dependencies here.
+# Include plugin headers
 target_include_directories(${PLUGIN_NAME} INTERFACE
-  "${CMAKE_CURRENT_SOURCE_DIR}/include")
-target_link_libraries(${PLUGIN_NAME} PRIVATE flutter)
-target_link_libraries(${PLUGIN_NAME} PRIVATE PkgConfig::GTK)
+  "${CMAKE_CURRENT_SOURCE_DIR}/include"
+)
 
-# List of absolute paths to libraries that should be bundled with the plugin.
-# This list could contain prebuilt libraries, or libraries created by an
-# external build triggered from this build file.
+# Link with flutter and GTK (if needed)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(GTK REQUIRED gtk+-3.0)
+
+target_include_directories(${PLUGIN_NAME} PRIVATE
+  ${GTK_INCLUDE_DIRS}
+)
+
+target_compile_options(${PLUGIN_NAME} PRIVATE ${GTK_CFLAGS_OTHER})
+
+target_link_libraries(${PLUGIN_NAME} PRIVATE
+  flutter
+  ${GTK_LIBRARIES}
+)
+
+# Bundled libraries placeholder
 set(disk_space_2_bundled_libraries
   ""
   PARENT_SCOPE

--- a/linux/CMakeLists.txt
+++ b/linux/CMakeLists.txt
@@ -1,25 +1,49 @@
-cmake_minimum_required(VERSION 3.14)
+# The Flutter tooling requires that developers have CMake 3.10 or later
+# installed. You should not increase this version, as doing so will cause
+# the plugin to fail to compile for some customers of the plugin.
+cmake_minimum_required(VERSION 3.10)
+
+# Project-level configuration.
 set(PROJECT_NAME "disk_space_2")
 project(${PROJECT_NAME} LANGUAGES CXX)
 
-set(PLUGIN_NAME "${PROJECT_NAME}_plugin")
+# This value is used when generating builds using this plugin, so it must
+# not be changed.
+set(PLUGIN_NAME "disk_space_2_plugin")
 
+# Any new source files that you add to the plugin should be added here.
+list(APPEND PLUGIN_SOURCES
+  "disk_space_2_plugin.cc"
+)
+
+# Define the plugin library target. Its name must not be changed (see comment
+# on PLUGIN_NAME above).
 add_library(${PLUGIN_NAME} SHARED
-  "${PLUGIN_NAME}.cc"
+  ${PLUGIN_SOURCES}
 )
 
+# Apply a standard set of build settings that are configured in the
+# application-level CMakeLists.txt. This can be removed for plugins that want
+# full control over build settings.
+apply_standard_settings(${PLUGIN_NAME})
+
+# Symbols are hidden by default to reduce the chance of accidental conflicts
+# between plugins. This should not be removed; any symbols that should be
+# exported should be explicitly exported with the FLUTTER_PLUGIN_EXPORT macro.
 set_target_properties(${PLUGIN_NAME} PROPERTIES
-  CXX_VISIBILITY_PRESET hidden
-  VERSION 1.0.0
-  SOVERSION 1
-)
-
+  CXX_VISIBILITY_PRESET hidden)
 target_compile_definitions(${PLUGIN_NAME} PRIVATE FLUTTER_PLUGIN_IMPL)
+
+# Source include directories and library dependencies. Add any plugin-specific
+# dependencies here.
 target_include_directories(${PLUGIN_NAME} INTERFACE
   "${CMAKE_CURRENT_SOURCE_DIR}/include")
 target_link_libraries(${PLUGIN_NAME} PRIVATE flutter)
 target_link_libraries(${PLUGIN_NAME} PRIVATE PkgConfig::GTK)
 
+# List of absolute paths to libraries that should be bundled with the plugin.
+# This list could contain prebuilt libraries, or libraries created by an
+# external build triggered from this build file.
 set(disk_space_2_bundled_libraries
   ""
   PARENT_SCOPE

--- a/linux/CMakeLists.txt
+++ b/linux/CMakeLists.txt
@@ -1,38 +1,24 @@
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.14)
 set(PROJECT_NAME "disk_space_2")
 project(${PROJECT_NAME} LANGUAGES CXX)
-
-# Add compiler warnings and optimizations
-if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-  target_compile_options(${PROJECT_NAME} PRIVATE
-    -Wall
-    -Wextra
-    -Wpedantic
-    -Werror=return-type
-    -O2
-  )
-endif()
 
 set(PLUGIN_NAME "${PROJECT_NAME}_plugin")
 
 add_library(${PLUGIN_NAME} SHARED
   "${PLUGIN_NAME}.cc"
 )
-apply_standard_settings(${PLUGIN_NAME})
+
 set_target_properties(${PLUGIN_NAME} PROPERTIES
   CXX_VISIBILITY_PRESET hidden
   VERSION 1.0.0
   SOVERSION 1
 )
+
 target_compile_definitions(${PLUGIN_NAME} PRIVATE FLUTTER_PLUGIN_IMPL)
 target_include_directories(${PLUGIN_NAME} INTERFACE
   "${CMAKE_CURRENT_SOURCE_DIR}/include")
 target_link_libraries(${PLUGIN_NAME} PRIVATE flutter)
 target_link_libraries(${PLUGIN_NAME} PRIVATE PkgConfig::GTK)
-
-# Add documentation
-install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/README.md"
-  DESTINATION "${CMAKE_INSTALL_DOCDIR}/${PROJECT_NAME}")
 
 set(disk_space_2_bundled_libraries
   ""

